assets/js/modules/search-console/components/module/ModuleOverviewWidget/Overview.js

```diff
@@ -0,0 +1,194 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import each from 'lodash/each';
+import round from 'lodash/round';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { calculateChange } from '../../../../../util';
+import { Grid, Row, Cell } from '../../../../../material-components';
+import DataBlock from '../../../../../components/DataBlock';
+
+const reduceSearchConsoleData = ( rows ) => {
+	const dataMap = [
+		[
+			{ type: 'string', label: __( 'Day', 'google-site-kit' ) },
+			{ type: 'number', label: __( 'Clicks', 'google-site-kit' ) },
+			{ type: 'number', label: __( 'Impressions', 'google-site-kit' ) },
+			{ type: 'number', label: __( 'CTR', 'google-site-kit' ) },
+			{ type: 'number', label: __( 'Position', 'google-site-kit' ) },
+		],
+	];
+
+	let totalClicks = 0;
+	let totalImpressions = 0;
+	let totalCTR = 0;
+	let totalPosition = 0;
+	const count = rows.length;
+
+	each( rows, ( row ) => {
+		const date = new Date( row.keys[ 0 ] );
+		dataMap.push( [
+			( date.getMonth() + 1 ) + '/' + date.getUTCDate(),
+			row.clicks,
+			row.impressions,
+			round( row.ctr, 3 ),
+			round( row.position, 3 ),
+		] );
+		totalClicks += row.clicks;
+		totalImpressions += row.impressions;
+		totalCTR += row.ctr;
+		totalPosition += row.position;
+	} );
+
+	// Do not divide by zero.
+	const averageCTR = count > 0 ? totalCTR / count : 0.0;
+	const averagePosition = count > 0 ? totalPosition / count : 0.0;
+
+	return {
+		dataMap,
+		totalClicks,
+		totalImpressions,
+		averageCTR,
+		averagePosition,
+	};
+};
+
+const extractSearchConsoleDashboardData = ( rows ) => {
+	// Split the results in two chunks.
+	const half = Math.floor( rows.length / 2 );
+	// Rows are from oldest to newest.
+	const latestData = reduceSearchConsoleData( rows.slice( half ) );
+	const olderData = reduceSearchConsoleData( rows.slice( 0, half ) );
+
+	return {
+		dataMap: latestData.dataMap,
+		totalClicks: latestData.totalClicks,
+		totalImpressions: latestData.totalImpressions,
+		averageCTR: latestData.averageCTR,
+		averagePosition: latestData.averagePosition,
+		totalClicksChange: calculateChange( olderData.totalClicks, latestData.totalClicks ),
+		totalImpressionsChange: calculateChange( olderData.totalImpressions, latestData.totalImpressions ),
+		averageCTRChange: calculateChange( olderData.averageCTR, latestData.averageCTR ),
+		averagePositionChange: calculateChange( olderData.averagePosition, latestData.averagePosition ),
+	};
+};
```

Comment:
Why are we duplicating these utilities?

https://github.com/google/site-kit-wp/blob/55c3280c1e61cbce8aa9360592c72a3fe452460b/assets/js/modules/search-console/util/index.js#L31-L74

https://github.com/google/site-kit-wp/blob/55c3280c1e61cbce8aa9360592c72a3fe452460b/assets/js/modules/search-console/util/index.js#L76-L94

Reply:
@aaemnnosttv Please see @felixarntz 's comment here: https://github.com/google/site-kit-wp/pull/3015#discussion_r600637259

Reply:
@asvinb That comment was meant to make this as a future enhancement, once we remove the legacy version of the component. We shouldn't duplicate the code for them (basically it's okay using the existing utility function as is for now).

Reply:
Updated @aaemnnosttv